pkg/api/docs/definitions/folder.go

```diff
@@ -0,0 +1,174 @@
+package definitions
+
+import (
+	"github.com/grafana/grafana/pkg/api/dtos"
+	"github.com/grafana/grafana/pkg/models"
+)
+
+// swagger:route GET /folders folders getFolders
+//
+// Get all folders.
+//
+// Returns all folders that the authenticated user has permission to view.
+//
+// Responses:
+// 200: getFoldersResponse
+// 401: unauthorisedError
+// 403: forbiddenError
+// 500: internalServerError
+
+// swagger:route GET /folders/{folder_uid} folders getFolderByUID
+//
+// Get folder by uid.
+//
+// Responses:
+// 200:
+// 401: unauthorisedError
+// 403: forbiddenError
+// 404: notFoundError
+// 500: internalServerError
+
+// swagger:route POST /folders folders createFolder
+//
+// Create folder.
+//
+// Responses:
+// 200: folderResponse
+// 400: badRequestError
+// 401: unauthorisedError
+// 403: forbiddenError
+// 409: conflictError
+// 500: internalServerError
+
+// swagger:route PUT /folders/{folder_uid} folders updateFolder
+//
+// Update folder.
+//
+// Responses:
+// 200: folderResponse
+// 400: badRequestError
+// 401: unauthorisedError
+// 403: forbiddenError
+// 404: notFoundError
+// 409: conflictError
+// 500: internalServerError
+
+// swagger:route DELETE /folders/{folder_uid} folders deleteFolder
+//
+// Delete folder.
+//
+// Deletes an existing folder identified by UID along with all dashboards (and their alerts) stored in the folder. This operation cannot be reverted.
+//
+// Responses:
+// 200: deleteFolderResponse
+// 400: badRequestError
+// 401: unauthorisedError
+// 403: forbiddenError
+// 404: notFoundError
+// 500: internalServerError
+
+// swagger:route GET /folders/id/{folder_id} folders getFolderByID
+//
+// Get folder by id.
+//
+// Returns the folder identified by id.
+//
+// Responses:
+// 200: folderResponse
+// 401: unauthorisedError
+// 403: forbiddenError
+// 404: notFoundError
+// 500: internalServerError
+
+// swagger:parameters getFolders
+type GetFoldersParams struct {
+	// Limit the maximum number of folders to return
+	// in:query
+	// required:false
+	// default:1000
+	Limit int64 `json:"limit"`
+	// Page index for starting fetching folders
+	// in:query
+	// required:false
+	// default:1
+	Page int64 `json:"page"`
+}
+
+// swagger:parameters getFolderByUID updateFolder deleteFolder
+// swagger:parameters getFolderPermissions updateFolderPermissions
+type FolderUIDParam struct {
+	// in:path
+	// required:true
+	FolderUID string `json:"folder_uid"`
+}
+
+// swagger:parameters getFolderByID
+type FolderIDParam struct {
+	// in:path
+	// required:true
+	FolderID int64 `json:"folder_id"`
+}
+
+// swagger:parameters createFolder
+type CreateFolderParam struct {
+	// in:body
+	// required:true
+	Body models.CreateFolderCommand `json:"body"`
```

Comment:
there are two way to resolve this:
- add the missing `json:"-"` from the [Result](https://github.com/grafana/grafana/blob/d21abdfe771b290060d3009c8ec05c17e15e497e/pkg/models/folders.go#L65) field or
- add an `//swagger:ignore` comment

even though I believe that the former is more correct, I tend towards the latter since that applies only in the swagger spec while the other change has a larger scope.

Reply:
Good catch :+1: Thought we already had `json:"-"` for all fields that we're not allowing. Could potentially be fixed after this has been merged, but up to you.